jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3ViewerPanel.java

```diff
@@ -3,88 +3,91 @@
+    private final ColumnInfo<Object, String>[] columns;
+    private S3VirtualBucket virtualBucket;
 
     public S3ViewerPanel(Disposable disposable, Project project, S3VirtualBucket bucketVirtual) {
-        this.disposable = disposable;
         this.project = project;
-
-        name.setText(bucketVirtual.getName());
-        date.setText(S3Resources.formatDate(bucketVirtual.getS3Bucket().creationDate()));
-
-        AwsRegion activeRegion = ProjectAccountSettingsManager.getInstance(project).getActiveRegion();
-        arnText.setText(S3UtilsKt.bucketArn(bucketVirtual.getName(), activeRegion));
-
-        s3TreeNode = new S3TreeDirectoryNode(bucketVirtual, null, "");
+        this.disposable = disposable;
+        this.virtualBucket = bucketVirtual;
 
         ColumnInfo<Object, String> key = new S3Column(S3ColumnType.NAME);
         ColumnInfo<Object, String> size = new S3Column(S3ColumnType.SIZE);
         ColumnInfo<Object, String> modified = new S3Column(S3ColumnType.LAST_MODIFIED);
-        ColumnInfo<Object, String>[] columns = new ColumnInfo[] {key, size, modified};
-        model = createTreeTableModel(columns);
-
-        DefaultTableCellRenderer tableRenderer = new DefaultTableCellRenderer();
-        tableRenderer.setHorizontalAlignment(SwingConstants.LEFT);
-
+        columns = new ColumnInfo[] {key, size, modified};
+        S3TreeTableModel model = createTreeTableModel(columns, new S3TreeDirectoryNode(bucketVirtual, null, ""));
         treeTable = new S3TreeTable(model, bucketVirtual, project);
-        treeTable.setRootVisible(false);
-        treeTable.setDefaultRenderer(Object.class, tableRenderer);
-        S3TreeCellRenderer treeRenderer = new S3TreeCellRenderer(treeTable);
-        treeTable.setTreeCellRenderer(treeRenderer);
-        treeTable.setCellSelectionEnabled(false);
-
-        treeTable.setRowSelectionAllowed(true);
-        treeTable.setAutoCreateRowSorter(true);
-        treeTable.setRowSorter(new S3RowSorter(treeTable.getModel()));
+        applyTreeStyle(treeTable);
+        addTreeActions(treeTable);
 
-        addTreeActions();
+        ToolbarDecorator panel = addToolbar(treeTable);
+        mainPanel.add(panel.createPanel());
+    }
 
-        treeTable.getColumnModel().getColumn(1).setMaxWidth(120);
+    private void createUIComponents() {
+    }
 
-        mainPanel.add(ScrollPaneFactory.createScrollPane(treeTable), BorderLayout.CENTER);
+    private ToolbarDecorator addToolbar(S3TreeTable table) {
+        DefaultActionGroup group = makeActionGroup(table);
+        group.addAction(new AnAction(message("explorer.refresh.title"), null, AllIcons.Actions.Refresh) {
+            @Override
+            public boolean isDumbAware() {
+                return true;
+            }
+
+            @Override
+            public void actionPerformed(@NotNull AnActionEvent e) {
+                table.setModel(createTreeTableModel(columns, new S3TreeDirectoryNode(virtualBucket, null, "")));
```

Comment:
Why do we need a new model? This is using the same AsyncTreeModel as the explorer, why can't we invalidate it the same way?

Reply:
Yeah this was due to it not working, looked at it again today and it works fine.